### PR TITLE
DLS-9285: Content change for Work out Capital Gains Tax due page

### PR DESCRIPTION
**Pull request by johno1985**

### CODE REVIEW

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/repos/Repo.scala [hunk 1]
```diff
@@ -36,27 +34,7 @@ trait Repo {
...
-                              s"Could not parse session data from mongo: ${e.mkString("; ")}"
-                            )
-                          )
-                      )
-          } yield result
-
-          response.value
-        }
-        .recover { case e => Left(Error(e)) }
+      EitherT.liftF(cacheRepository.get(id)(DataKey(sessionKey))).value
```
Comment (LAJW) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/repos/Repo.scala:37]: 💡 You could be cheeky and avoid EitherT altogether:
```scala
      cacheRepository.get(id)(DataKey(sessionKey)).map(Right(_))
```

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/yeartodatelliability/YearToDateLiabilityController.scala [hunk 2]
```diff
@@ -1809,7 +1816,14 @@ class YearToDateLiabilityController @Inject() (
                           _,
                           fillingOutReturn.isAmendReturn,
                           fillingOutReturn.subscribedDetails.isATrust,
-                          wasUkResident
+                          wasUkResident,
+                          fillingOutReturn.draftReturn match {
```
Comment (LAJW) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/yeartodatelliability/YearToDateLiabilityController.scala:1820]: ⛏️ Could you use a named argument for this boolean?

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/yeartodatelliability/YearToDateLiabilityController.scala [hunk 3]
```diff
@@ -1737,7 +1737,14 @@ class YearToDateLiabilityController @Inject() (
...
```
Comment (LAJW) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/yeartodatelliability/YearToDateLiabilityController.scala:1741]: ⛏️ Could you use a named argument for this boolean?

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/returns/ytdliability/non_calculated_enter_tax_due.scala.html [hunk 4]
```diff
@@ -46,24 +53,38 @@
   else if(wasUkResident) ".ukResident" -> viewConfig.calculateCgt
   else ".nonResident" -> viewConfig.calculateCgtNr
 }
+@label = @{messages(s"$key.label")}
```
Comment (LAJW) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/returns/ytdliability/non_calculated_enter_tax_due.scala.html]: ⛏️ I'd prefer not assembling messages' keys from separate strings but rather use whole strings so they're easier to locate in messages.

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/repos/Repo.scala (hunk 1 above)
Comment (johno1985) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/repos/Repo.scala:37]: Yeah, I was thinking that based on what is below but completely forgot to try it. Will try that now

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/returns/ytdliability/non_calculated_enter_tax_due.scala.html (hunk 4 above)
Comment (johno1985) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/returns/ytdliability/non_calculated_enter_tax_due.scala.html]: I wanted to follow the existing pattern, but yes, I agree, using compound key expressions makes it more difficult to find references

**Review by LAJW — APPROVED** [review]

**Pull request merged**